includes/wikia/parser/templatetypes/handlers/NavigationTemplate.class.php

```diff
@@ -29,26 +29,37 @@ public static function resolve( &$html ) {
 	}
 
 	private static function process( $html ) {
-		$regex = '/<(' . implode( '|', self::$blockLevelElements ) . ')[>\s]+/i';
-		$marked = self::mark( "(.*)" );
-
-		preg_match_all( "/{$marked}/sU", $html, $matches );
-		foreach ( $matches[ 0 ] as $key => $found ) {
-			$replacement = $matches[ 1 ][ $key ];
-			if ( preg_match( $regex, $matches[ 1 ][ $key ] ) ) {
-				$replacement = '';
+		$blockElemRegex = '/<(' . implode( '|', self::$blockLevelElements ) . ')[>\s]+/i';
+		$markerRegex = "/\x7f".self::$mark.".+?\x7f/s";
+
+		//getting markers of each navigation template
+		preg_match_all($markerRegex, $html, $markers);
+		foreach ( $markers[ 0 ] as $marker ) {
+			$replacementRegex = '/'.$marker.".*?".$marker.'/s';
+			preg_match_all($replacementRegex, $html, $navTemplates);
+
+			//multiple invocations of the same template can occur, replacing each of them
+			foreach($navTemplates[0] as $navTemplate) {
+				$replacement = str_replace($marker, '', $navTemplate);
+
+				if ( preg_match( $blockElemRegex, $navTemplate ) ) {
+					$replacement = '';
+				}
+				$html = str_replace( $navTemplate, $replacement, $html );
 			}
-			$html = str_replace( $found, $replacement, $html );
+
```

Comment:
cool idea, although in my opinion regex for 2. should be:
2. preg_replace_all "marker.*(block elements).*marker"